web/static/js/actions/channels.js

```diff
@@ -26,31 +26,42 @@ export const receiveChannels = (response) => ({
   response
 })
 
-export const createNuntiumChannel = () => dispatch => {
-  return Promise.all([
-    authorizeWithGuisso(dispatch, 'nuntium', config.nuntium),
-    pigeon.loadPigeonScript(config.nuntium.baseUrl)
-  ])
-    .then(([token, _]) => pigeon.addChannel(token.access_token))
-    .then(nuntiumChannel => {
-      if (nuntiumChannel === null) {
-        return Promise.reject('User cancelled')
-      }
-      dispatch(createChannel({
-        name: nuntiumChannel.name,
-        type: 'sms',
-        provider: 'nuntium',
-        settings: {
-          nuntiumChannel: nuntiumChannel.name
+export const createNuntiumChannel = (() => {
+  let creatingChannel = false
+  return () => dispatch => {
+    if (creatingChannel) {
```

Comment:
This has the problem that if the authorization popup is closed, there is no way to start over the process. I guess this check must be done right before `addChannel`.

Also, in `guisso.js` we could use a fixed name for the window (instead of `_blank`) to avoid multiple popups.